radix-engine/src/system/system.rs

```diff
@@ -2858,21 +2858,56 @@ where
     Y: KernelApi<SystemConfig<V>>,
     V: SystemCallbackObject,
 {
-    #[trace_resources]
+    #[trace_resources(log=message.len())]
     fn bls12381_v1_verify(
         &mut self,
-        message: Vec<u8>,
-        public_key: Bls12381G1PublicKey,
-        signature: Bls12381G2Signature,
+        message: &[u8],
+        public_key: &Bls12381G1PublicKey,
+        signature: &Bls12381G2Signature,
     ) -> Result<u32, RuntimeError> {
         // TODO: apply execution costs
-        Ok(verify_bls12381_v1(&message, &public_key, &signature) as u32)
+        Ok(verify_bls12381_v1(message, public_key, signature) as u32)
     }
 
-    #[trace_resources]
-    fn keccak256_hash(&mut self, data: Vec<u8>) -> Result<Hash, RuntimeError> {
+    // Trace average message length and number of public_keys
+    // FIXME: Improve resource tracker, because below line causes panic in resource tracker
+    //#[trace_resources(log=(messages.iter().map(|m| m.len()).sum::<usize>()/messages.len()),log=public_keys.len())]
```

Comment:
`messages` is a vector of vectors.
We need to have a proper way to determine it's total size in order to properly cost it.

Reply:
Fixed